include/cantera/zeroD/Wall.h

```diff
@@ -1,35 +1,40 @@
-//! @file Wall.h Header file for class Wall.
+//! @file Wall.h Header file for base class WallBase.
 
 // This file is part of Cantera. See License.txt in the top-level directory or
-// at http://www.cantera.org/license.txt for license and copyright information.
+// at https://cantera.org/license.txt for license and copyright information.
 
 #ifndef CT_WALL_H
 #define CT_WALL_H
 
 #include "cantera/base/ctexceptions.h"
-#include "cantera/numerics/Func1.h"
-#include "cantera/zeroD/ReactorBase.h"
 #include "cantera/zeroD/ReactorSurface.h"
+#include "cantera/zeroD/ReactorBase.h"
+#include "cantera/zeroD/DeviceConst.h"
 
 namespace Cantera
 {
 
 class Kinetics;
 class SurfPhase;
+class Func1;
 
-//! Represents a wall between between two ReactorBase objects.
-/*!
- * Walls can move (changing the volume of the adjacent reactors) and allow heat
- * transfer between reactors.
+/**
+ * Base class for 'walls' (walls, pistons, etc.) connecting reactors.
+ * @ingroup reactor0
  */
-class Wall
+class WallBase
```

Comment:
I think some more clarity is needed on the separation between `WallBase` and `Wall`. I recognize that introducing a base class with fewer "features" may make adding new derived classes simpler, but I don't fully follow the logic of what is included in `WallBase` versus `Wall`. Can you please elaborate?

Reply:
The distinction is arbitrary, but I wanted to mimic `ReactorBase` vs `Reactor` in the `WallBase`/`Wall` area. From my perspective, the `Func1` is a specialization for a multi-purpose object communicating with Python, which may not be necessary for another case. There are many other ways to differentiate. Suggestions are welcome.

Reply:
To elaborate further: anyone interested in changing the behavior of wall objects will likely overload heat transfer (`WallBase::qdot`) and wall velocity (`WallBase::vdot`), which is what motivated the proposed separation.

Reply:
Right, making `vdot` and `qdot` virtual certainly makes sense. I don't really see the use of `Func1` as being applicable only to Python -- it's meant to be used any time you want to set some parameter as a function of time. That interface could probably be revised to make use of C++11 lambdas rather than the custom `Func1` class, but the concept would remain the same.

I guess you could move all the functions related to heat transfer and motion away from the base wall class. That doesn't leave much left in `WallBase`, but it would mean that you'd have a pretty clean and clear interface for any new derived classes to be built on.

Reply:
Good suggestion. It is now implemented.